src/craftdef.cpp

```diff
@@ -960,6 +960,81 @@ class CCraftDefManager: public IWritableCraftDefManager
 
 		return recipes;
 	}
+
+	virtual bool clearCraftRecipesByOutput(CraftOutput &output, IGameDef *gamedef)
+	{
+		std::map<std::string, std::vector<CraftDefinition*> >::iterator vec_iter = m_output_craft_definitions.find(output.item);
+
+		if (vec_iter == m_output_craft_definitions.end())
+			return false;
+
+		std::vector<CraftDefinition*> &vec = vec_iter->second;
+		for (std::vector<CraftDefinition*>::iterator i = vec.begin();
+			i != vec.end(); ++i) {
+			CraftDefinition *def = *i;
+			std::vector<CraftDefinition*> &unhashed_inputs_vec = m_craft_defs[(int) CRAFT_HASH_TYPE_UNHASHED][0]; // Recipes are not yet hashed at this point
+			std::vector<CraftDefinition*> new_vec_by_input;
```

Comment:
Can you add a `new_vec_by_input.reserve(unhashed_inputs_vec.size())` ? Its a clear upper bound for `new_vec_by_input`'s size and avoids unneccessary reallocations.